src/dbnode/storage/index_test.go

```diff
@@ -156,6 +156,150 @@ func TestNamespaceIndexCleanupDuplicateFilesets(t *testing.T) {
 	require.NoError(t, idx.CleanupDuplicateFileSets())
 }
 
+func TestNamespaceIndexCleanupDuplicateFilesets_GroupingByBlockStartAndVolume(t *testing.T) {
```

Comment:
This test is unrelated to the change. I added it to improve coverage of grouping logic in the cleanup method.

Reply:
Is it by `Volume`, or by `VolumeType`?

Reply:
As I understand, they are grouped by block start and volume type (which the test name is supposed to reflect?), and within each such group they are ordered by volume index (which is perhaps just a cleanup logics implementation detail with no goal of being tested here)?
The essence being that cleanup decisions are being scoped to `BlockStartAndVolumeType`?

Reply:
https://github.com/m3db/m3/pull/3409/commits/71ba88577cb6eea97786e4cef6a5f27b816b65bd